streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamPartitionAssignorTest.java

```diff
@@ -798,17 +798,43 @@ public void shouldReturnEmptyClusterMetadataIfItHasntBeenBuilt() throws Exceptio
     }
 
     @Test
-    public void shouldNotLoopInfinitelyOnMissingMetadataAndShouldNotCreateRelatedTasks() {
+    public void shouldCreateAllTasksAndInternalTopicIfAllInputTopicsAreKnown() {
+        runMetadataTest("topic1", true, "topic3", true);
+    }
+
+    @Test
+    public void shouldNotLoopInfinitelyOnPartlyMissingMetadataAndShouldNotCreateRelatedTasks1() {
```

Comment:
What's a bit odd about this test and others like `shouldNotLoopInfinitelyOnPartlyMissingMetadataAndShouldNotCreateRelatedTasks2` is that the test name talks about many things (and many different things, across the various tests), but ultimately the test is just a single call to `runMetadataTest`.